### PR TITLE
Add Marathon Petroleum (US) to replace marathon and arco

### PR DESCRIPTION
**Pull request by davidhicks**
Marathon Petroleum owns the Marathon and Arco brands, and the same store finder is used for both brands. Currently the `marathon` and `arco` spiders are broken. These old spiders are now replaced by the single Marathon Petroleum spider this commit introduces.

Fuel types are very confusing and there are 100's of different codes applying to different types of fuels. The matching between codes and OSM fuel tag values is best-effort.

### CODE REVIEW

File: locations/spiders/marathon_petroleum_us.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+                        fuel_types_matched.append(Fuel.OCTANE_87)
+                    case "UNLD88":
+                        fuel_types_matched.append(Fuel.OCTANE_88)
```
Comment (iandees) [inline locations/spiders/marathon_petroleum_us.py:50]: Looks like `Fuel.OCTANE_88` might need to get added

Comment (davidhicks) [reply locations/spiders/marathon_petroleum_us.py:50]: Thanks, I lost that change to categories.py with an accidental git command. Added 88 and also 96 that was missing (completing the full list of 87 through 100 inclusive)

Comment (iandees) [reply locations/spiders/marathon_petroleum_us.py:50]: Thanks for going through all the fuel types. That's a lot of distinct values!

**Comment by iandees** [comment]
Wonder why it's only getting one on Marathon. I see many more in the start url.

**Comment by davidhicks** [comment]
> Wonder why it's only getting one on Marathon. I see many more in the start url.

Doh I forgot to add back `yield item`! I removed it during testing

**Comment by davidhicks** [comment]
> Wonder why it's only getting one on Marathon. I see many more in the start url.

I'm missing the `ref` field... fixing it

**Comment by davidhicks** [comment]
Fixed. I had also accidentally lost that change to DictParser during testing. Hopefully I've got everything now included :)

**Pull request merged**